src/server/json_family.cc

```diff
@@ -1485,7 +1464,14 @@ OpStatus OpMerge(const OpArgs& op_args, string_view key, std::string_view json_s
     op_args.shard->search_indices()->RemoveDoc(key, op_args.db_cntx, it_res->it->second);
 
     JsonType* obj = it_res->it->second.GetJson();
-    RecursiveMerge(*parsed_json, obj);
+    try {
+      // https://datatracker.ietf.org/doc/html/rfc7386#section-2
+      mergepatch::apply_merge_patch(*obj, *parsed_json);
```

Comment:
@romange I am 99% sure this doesn't throw but I will keep the `try/block` just in case. + I don't think we now need to worry about correctness.

Reply:
nice! Did not that they had this API 💯